lib/ProductOpener/Food.pm

```diff
@@ -4820,8 +4909,39 @@ sub compute_serving_size_data($) {
 	#	$product_ref->{nutriments}{'energy.unit'} = 'kj';
 	#}
 
+
 	foreach my $product_type ("", "_prepared") {
 
+		# Energy
+		# Before November 2019, we only had one energy field with an input value in kJ or in kcal, and internally it was converted to kJ
+		# In Europe, the energy is indicated in both kJ and kcal, but there isn't a straightforward conversion between the 2: the energy is computed
+		# by summing some nutrients multiplied by an energy factor. That means we need to store both the kJ and kcal values.
+		# see bug https://github.com/openfoodfacts/openfoodfacts-server/issues/2396
+
+		# If we have a value for energy-kj, use it for energy
+		if (defined $product_ref->{nutriments}{"energy-kj" . $product_type}) {
```

Comment:
So we give priority to kJ?

Reply:
Yes, kJ is the official value in EU, kcal is historic. In the US we only have kcal. Also kJ is more precise.

Reply:
Perfect